src/coordinator/mod.rs

```diff
@@ -14,11 +15,25 @@ pub type InputsStore = std::collections::HashMap<Serial, lxp::packet::ReadInputs
 pub struct Coordinator {
     config: ConfigWrapper,
     channels: Channels,
+
+    inverter_receiver: std::cell::Cell<Option<inverter::Receiver>>,
```

Comment:
The cell gives us interior mutability; the option gives us the ability to swap out the mutable receiver and put a `None` in its place instead.

There might be better ways to achieve the same thing; the critical aspect of this design is that the receiver is created in the initializer, rather than as a result of the `start()` method.

Reply:
Bit of a shame about the added complexity. I wonder if there's a case for merging `new` and `start`? I'm not sure if there's any reason we couldn't really. So just creating an instance is enough to get things running; maybe it would be a bit weird for `new` to return a future though..

Reply:
OK, I've put `Cell` in the prelude.

Merging `new` and `start` would bring back the same problem - we'd be creating receivers in `newstart` in some components while other components would be trying to send to them in their own `newstart`s. Moving the receiver creation to `new` is what makes this work :sweat_smile:

I think using `Cell` is quite normal in this case, but the `Option` is a bit yucky, for sure. It has the benefit of being a small change with the existing code.

It would be a bigger refactor, but we could get rid of the `Option`s by moving the code in the respective `stop()` methods to a drop trait (or otherwise doing away with it), then changing `start` to be `&mut self`. We'd then be able to use `Cell.into_inner()` instead of `replace()` to access the receiver, I think.

Reply:
Its a very minor thing, but I see `Cell` has a `take` that replaces the value with `default`, which would be `None`: https://doc.rust-lang.org/stable/std/option/enum.Option.html#impl-Default-for-Option%3CT%3E

Think it'd be a nice little cleanup to use that instead.

After that I'm happy to merge this.

Reply:
Great :thumbsup: - I saw it, but before I'd realised I needed the `Option` :sweat_smile:. I've made that change now.